src/elements/dropdown/dropdown.js

```diff
@@ -0,0 +1,455 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('semantic.ui.elements.dropdown', [])
+    .directive('smDropdown', smDropdown)
+    .directive('smDropdownItem', smDropdownItem)
+    .directive('smDropdownSearch', smDropdownSearch);
+
+  var keys = {
+    enter      : 13,
+    space      : 32,
+    escape     : 27,
+    leftArrow  : 37,
+    upArrow    : 38,
+    rightArrow : 39,
+    downArrow  : 40
+  };
+
+  function smDropdown() {
+    return {
+      restrict: 'E',
+      require: ['smDropdown', '?ngModel'],
+      compile: compile,
+      controller: DropdownCtrl
+    };
+
+    function compile(element, attrs) {
+      var
+        labelElm = element.find('sm-dropdown-label'),
+        menuElm = element.find('sm-dropdown-menu');
+
+      labelElm.addClass('text');
+
+      menuElm.addClass('menu transition');
+      menuElm.attr('tabindex', -1);
+
+      element.addClass('ui dropdown');
+      if (angular.isDefined(attrs.type)) {
+        element.addClass(attrs.type);
+      }
+
+      if (isType('selection')) {
+        labelElm.addClass('default');
+      }
+
+      if (isType('search')) {
+        element.append('<input class="search" tabindex="0" sm-dropdown-search>');
+      } else {
+        element.attr('tabindex', 0);
+      }
+
+      return postLink;
+
+      function isType(type) {
+        return attrs.type && attrs.type.indexOf(type) > -1;
+      }
+    }
+
+    function postLink(scope, element, attrs, ctrls) {
+      var
+        dropdownCtrl = ctrls[0],
+        ngModel = ctrls[1];
+
+      dropdownCtrl.init(ngModel);
+    }
+  }
+
+  DropdownCtrl.$inject = ['$scope', '$element', '$attrs', '$parse', '$timeout', '$document'];
+
+  function DropdownCtrl($scope, $element, $attrs, $parse, $timeout, $document) {
+    var self = this,
+      mouseClick;
+
+    self.$scope = $scope;
+    self.$element = $element;
+    self.items = [];
+
+    self.labelElm = $element.find('sm-dropdown-label');
+
+    if ($attrs.isOpen) {
+      var getter = $parse($attrs.isOpen);
+      self.getIsOpen = function() {
+        return getter($scope);
+      };
+      self.setIsOpen = function(value) {
+        getter.assign($scope, value);
+      };
+    }
+
+    var menuElm = $element.find('sm-dropdown-menu');
+
+    $scope.$watch(function() {
+      return self.getIsOpen();
+    }, function(isOpen) {
+      $element[isOpen ? 'addClass' : 'removeClass']('active visible');
+
+      menuElm[isOpen ? 'addClass' : 'removeClass']('visible');
+      menuElm[!isOpen ? 'addClass' : 'removeClass']('hidden');
+
+      if (isOpen) {
+        var selectedItem = self.getSelectedItem() || self.items[0];
+
+        if (selectedItem) {
+          selectedItem.select();
+        }
+
+        $timeout(function() {
+          $document.on('click', closeDropdown);
```

Comment:
why would we need to listen on document's click and not just close dropdown on select?